[REFACTOR] 텍스트 라벨 컴포넌트 분리 (#289)

seuriseuljjeok opened the pull request:
# 🔥*Pull requests*

## ⛳ **작업한 브랜치**
- refactor/#289 

## 👷 **작업한 내용**
<!-- 작업한 내용을 적어주세요. -->
- [x] 텍스트 라벨 컴포넌트 분리

## 🚨 참고 사항
<!-- 참고할 사항이 있다면 적어주세요. -->
컴포넌트 관련 자세한 설명은 노션 참고해주세요!
https://good-fenugreek-191.notion.site/1c3fda2ab77b80d0a4e0d8fa4a957232?pvs=4


## 📟 관련 이슈
- Resolved: #289

seuriseuljjeok committed:
[Add] DRTextLabel, TextLabelType 추가 (#289)

seuriseuljjeok committed:
[Del] DRErrorLabel 삭제 (#289)

seuriseuljjeok committed:
[Feat] 텍스트 라벨 공용 컴포넌트  '일정 등록' 적용 (#289)

seuriseuljjeok committed:
[Feat] 텍스트 라벨 공용 컴포넌트  '코스 등록' 적용 (#289)

seuriseuljjeok committed:
[Feat] 텍스트 라벨 공용 컴포넌트  '배너' 적용 (#289)

seuriseuljjeok committed:
[Feat] 텍스트 라벨 공용 컴포넌트  '코스 둘러보기' 적용 (#289)

seuriseuljjeok committed:
[Feat] 텍스트 라벨 공용 컴포넌트  '코스 상세' 적용 (#289)

seuriseuljjeok committed:
[Feat] 텍스트 라벨 공용 컴포넌트  '데이트 일정/상세' 적용 (#289)

seuriseuljjeok committed:
[Feat] 텍스트 라벨 공용 컴포넌트  '메인' 적용 (#289)

seuriseuljjeok committed:
[Feat] 텍스트 라벨 공용 컴포넌트  '내가 열람한/등록한 코스' 적용 (#289)

seuriseuljjeok committed:
[Feat] 텍스트 라벨 공용 컴포넌트  '마이페이지' 적용 (#289)

seuriseuljjeok committed:
[Feat] 텍스트 라벨 공용 컴포넌트  '온보딩' 적용 (#289)

seuriseuljjeok committed:
[Feat] 텍스트 라벨 공용 컴포넌트 '포인트 상세' 적용 (#289)

seuriseuljjeok committed:
[Feat] 텍스트 라벨 공용 컴포넌트  '포인트 시스템' 적용 (#289)

seuriseuljjeok committed:
[Feat] 텍스트 라벨 공용 컴포넌트  '프로필 등록/편집' 적용 (#289)

seuriseuljjeok committed:
[Feat] 텍스트 라벨 공용 컴포넌트 다른 컴포넌트들에 적용 (#289)

seuriseuljjeok committed:
[Fix] 코스 상세 상단 네비바 버튼 색상 수정 (#289)

cirtuare commented on DATEROAD-iOS/DATEROAD-iOS/Presentation/DateSchedule/View/PastDateDetailViewController.swift:
```diff
@@ -112,30 +109,48 @@ extension PastDateDetailViewController: DRCustomAlertDelegate {
+    func didTapBottomButton() {
+        self.bottomSheetVC.dismissBottomSheet()
     }
+
+//    func didTapFirstLabel() {
+//        self.dismiss(animated: false)
+//        tapDeleteLabel()
+//    }
```
안 쓰는 거라면 지워도 될 것 같습니다 !!

seuriseuljjeok replied:
DRBottomSheetDelegate 에서 안쓰는 메소드들이 있는 것 같은데, 이거 관련 리팩은 우선 스프린트 기능 구현 후에 수정하려고 우선은 주석처리 해두었습니다!

cirtuare commented on DATEROAD-iOS/DATEROAD-iOS/Presentation/DateSchedule/View/UpcomingDateDetailViewController.swift:
```diff
@@ -258,22 +274,11 @@ extension UpcomingDateDetailViewController: DRBottomSheetDelegate {
-        self.dismiss(animated: false)
-        tapDeleteLabel()
-    }
+//    @objc
+//    func didTapFirstLabel() {
+//        self.dismiss(animated: false)
+//        tapDeleteLabel()
+//    }
```
요것도 안 쓰는 거라면 지워도 될 것 같아요 !!

cirtuare commented on DATEROAD-iOS/DATEROAD-iOS/Global/Utils/Enums/TextLabelType.swift:
```diff
@@ -0,0 +1,186 @@
+        case .extra20_gray300, .bold18_gray300, .bold11_gray300, .med15_gray300, .med13_gray300, .reg11_gray300:
+            return UIColor(.gray300)
+            
+        case .semi15_deepPurple, .reg11_purple600:
+            return UIColor(.deepPurple)
+            
+        case .med15_purple400:
+            return UIColor(.lightPurple)
```
제가 전 피알에서  lightPurple → purple400, mediumPurple → purple500, darkPurple → purple600 로 컬러 에셋 이름을 바꿔서 refactoring을 이 브랜치로 머지하면 여기서 오류날 것 같슴당 ..!!
얼렁 머지했어야 하는데 죄송함다 ㅠㅠ

cirtuare commented on DATEROAD-iOS/DATEROAD-iOS/Global/Utils/Enums/TextLabelType.swift:
```diff
@@ -0,0 +1,186 @@
+    
+    var bgColor: UIColor {
+        switch self {
+        case .deepPurple_12, .deepPurple_10:
+            return UIColor(.deepPurple)
+            
+        case .mediumPurple_14, .mediumPurple_10:
+            return UIColor(.mediumPurple)
```
여기두 컬러 에셋 명칭 변경 때문에 오류날 것 가타요
귀찮긴 하지만 컬러 에셋 이름 바꾼 김에 gray처럼 deepPurple_12 -> purple600_12로 바꾸면 통일성 있을 것 같습니다!

cirtuare reviewed (approved):
엄마 최고 !!! 짱깔끔하다 🌻💛

ParkSY0919 commented on DATEROAD-iOS/DATEROAD-iOS/Global/UIComponents/DRErrorViewController.swift:
```diff
@@ -13,9 +13,17 @@ final class DRErrorViewController: BaseNavBarViewController {
     private let errorImageView: UIImageView = UIImageView()
     
-    private let mainErrorMessageLabel: UILabel = UILabel()
+    private let mainErrorMessageLabel: DRTextLabel = DRTextLabel(
+        title: StringLiterals.Network.mainErrorMessage,
+        textLabelType: .clear(.extra20_gray300),
+        numberOfLines: 1
+    )
```
저희 정렬 방식 아래와 같이 아니었나유?!
```
...Label: DRTextLabel = DRTextLabel(title: StringLiterals.Network.mainErrorMessage,
                                                             textLabelType: .clear(.extra20_gray300),
                                                             numberOfLines: 1)
```

ParkSY0919 reviewed (approved):
어머니 바꾸신 파일만 봐도 Label이 얼마나 많이 쓰였는지 제가 감히 공감되려합니다..
더불어 `setNeedsDisplay()`로 Label 패딩 컨트롤 하시는 것 잘 보았습니다..!
고생 많으셨습니다!!!🔥

저희 매개변수 3개 이상인 경우의 정렬 방식만 다시 논의해서 잡고 가면 좋을 것 같습니다요!

seuriseuljjeok committed:
[Fix] conflict 해결 (#289)

# Conflicts:
#	DATEROAD-iOS/DATEROAD-iOS.xcodeproj/project.pbxproj
#	DATEROAD-iOS/DATEROAD-iOS/Global/Skeleton/UpcomingDateScheduleSkeletonView.swift
#	DATEROAD-iOS/DATEROAD-iOS/Global/UIComponents/DRTimelineView.swift
#	DATEROAD-iOS/DATEROAD-iOS/Presentation/Banner/Views/Cells/BannerInfoHeaderView.swift
#	DATEROAD-iOS/DATEROAD-iOS/Presentation/CourseDetail/Cells/TimelineInfoCell.swift
#	DATEROAD-iOS/DATEROAD-iOS/Presentation/CourseDetail/Views/ContentMaskView.swift
#	DATEROAD-iOS/DATEROAD-iOS/Presentation/DateSchedule/View/Cells/DateCardCollectionViewCell.swift
#	DATEROAD-iOS/DATEROAD-iOS/Presentation/DateSchedule/View/DateDetailContentView.swift
#	DATEROAD-iOS/DATEROAD-iOS/Presentation/DateSchedule/View/DateScheduleDeleteView.swift
#	DATEROAD-iOS/DATEROAD-iOS/Presentation/Main/Views/Cells/HotDateCourseCell.swift
#	DATEROAD-iOS/DATEROAD-iOS/Presentation/Main/Views/DateTicketView.swift
#	DATEROAD-iOS/DATEROAD-iOS/Presentation/Main/Views/EmptyTicketView.swift

seuriseuljjeok committed:
[Fix] 변경된 컬러 네이밍 적용 (#289)
